WTG.Analyzers/Analyzers/Var/VarAnalyzer.cs

```diff
@@ -123,6 +128,81 @@ static void AnalyzeInvoke(SyntaxNodeAnalysisContext context, FileDetailCache cac
 			}
 		}
 
+		static void AnalyzeAssignment(SyntaxNodeAnalysisContext context, FileDetailCache cache)
+		{
+			if (cache.IsGenerated(context.SemanticModel.SyntaxTree, context.CancellationToken))
+			{
+				return;
+			}
+
+			var node = (AssignmentExpressionSyntax)context.Node;
+
+			if (node.Left.IsKind(SyntaxKind.TupleExpression))
+			{
+				var type = context.SemanticModel.GetTypeInfo(node.Right).Type;
+
+				if (type != null && type.IsTupleType)
+				{
+					CheckTupleTypes(ref context, (TupleExpressionSyntax)node.Left, (INamedTypeSymbol)type);
+				}
+			}
+		}
+
+		static void CheckTupleTypes(ref SyntaxNodeAnalysisContext context, TupleExpressionSyntax tupleExp, INamedTypeSymbol type)
+		{
+			var arguments = tupleExp.Arguments;
+			var elements = type.TupleElements;
+
+			if (elements == null)
+			{
+				return;
+			}
+
+			var count = Math.Min(arguments.Count, elements.Length);
+
+			for (var i = 0; i < count; i++)
+			{
+				var arg = arguments[i].Expression;
+
+				switch (arg.Kind())
+				{
+					case SyntaxKind.TupleExpression:
+						{
```

Comment:
Hmm, maybe we should have an analyzer for case blocks too 😄 

I usually do:

```cs
case TheCase:
{
    // some code
    break;
}
```

rather than:

```cs
case TheCase:
    {
        // some code
    }
    break;
```

which is not IDE default for the braces, but I find it reads better. Do we have an implicit standard already in this project, or do we have mixed styles?

Reply:
For the most part, I tend to prefer putting the break outside the block. it feels more consistent (close the block before "closing the case".)

Where this falls apart is when the case is terminated by a return statement, which will almost certainly need to be inside the block.

Reply:
Any opinion on the indentation?

Reply:
In the absence of a broad consensus, I would stick with the default behaviour (the second example). Most (all?) of our existing code base is using this approach.